libs/io/tests/test_h5.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# coding: utf-8
+import os
+import shutil
+from pathlib import Path
+
+from typing import TYPE_CHECKING, Optional, Tuple
+import h5py
+import pytest
+import numpy as np
+
+
+from bbhnet.io import h5
+
+#from h5 import write_timeseries
+#from h5 import read_timeseries
+
+@pytest.fixture(scope="session")
+def write_dir():
+    write_dir = "tmp"
+    os.makedirs(write_dir, exist_ok=True)
+    yield Path(write_dir)
+    shutil.rmtree(write_dir)
+
+t = [1238163456,12345677890, 12345677890, 12345677890, 12345677890, 1238167552]
+y = [23, 53.5, 234, 0, 5]
+datasets = { 'dataset1': {'field_1': 'test1', 'field_2': 'test2', 'field_3': 'test3'},
```

Comment:
The `datasets` that get passed to `write` are meant to be numpy arrays, not strings. This is probably causing some of the issues that made changing the code necessary

Reply:
How about the keys? There are string keys, right? Would the following be a good representative for testing:                      

datasets = { 'dataset1': {'field_1': 2345, 'field_2': 45.25, 'field_3': 345567677},
                    'dataset2': {'field_1': 2, 'field_2': 45.4525, 'field_3': 3455673442.45} } ...?

If not, can you type out one I can use for testing that is as close to reality as possible?

I will wait on the rest of your comments until I resolve this one. This could be it.

Reply:
Yes the dataset fields are meant to be strings, but the values should be numpy arrays (not `dict`s). For that matter, so should `t` and `y`.

```python
t = np.array([...])
y = np.array([...])
datasets = {"dataset1": np.arange(len(t)), "dataset2": np.arange(len(t)) + 1 ...}
```

Couple other things to note is that all of these arrays should be of the same length, since the implication is that they represent values of some observable at each timestep in `t`. It's probably worth including an explicit check for this in the `write_timeseries` function itself that raises a `ValueError` if the shape is off, and including a test case that verifies if a `y` or `dataset` of the wrong length gets passed, this `ValueError` gets raised.

It's also worth including checks for verifying that things get passed appropriately when `write_timeseries` is called without the `**datasets` syntax, e.g.
```python
write_timeseries(t=t, y=y, dataset1=datasets["dataset1"])
```

Reply:
I think I understand this for the most part. Can you clarify  what does " + 1" mean in following context:

 datasets = {"dataset1": np.arange(len(t)), "dataset2": np.arange(len(t)) + 1 ...}

(Copied from your comment above)

Reply:
I made changed to the datasets object for your review.